ui/lib/kv/addon/components/page/list.js

```diff
@@ -38,7 +38,14 @@ export default class KvListPageComponent extends Component {
   }
 
   get buttonText() {
-    return pathIsDirectory(this.secretPath) ? 'View list' : 'View secret';
+    // secretPath is modified on typing.
+    if (this.secretPath) {
+      return pathIsDirectory(this.secretPath) ? 'View list' : 'View secret';
+    } else {
+      // if secretPath is an empty string it could be because the user hit a permissions error.
+      // in that case check pathToSecret, which is taken from the URL.
+      return pathIsDirectory(this.args.pathToSecret) ? 'View list' : 'View secret';
+    }
```

Comment:
could this be: 
```suggestion
  // if secretPath is an empty string it could be because the user hit a permissions error.
const path = this.secretPath || this.args.pathToSecret
return pathIsDirectory(path) ? 'View list' : 'View secret';
```

Reply:
yes yes it could. nice!